pkg/controller/utils/immutable_image.go

```diff
@@ -0,0 +1,42 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/apis/dynatrace/v1alpha1"
+	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/dtclient"
+	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/version"
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"time"
+)
+
+const updateInterval = 5 * time.Minute
+
+type Operator interface {
```

Comment:
it is used to check if the last check has already happened in the past five minutes. Since the status is changed in the `SetUseImmutableImageStatus` function, the code triggers an instant reconcile and without this check it would end up in an  endless cycle of setting state and instantly reconciling